api_pagopa.yaml

```diff
@@ -0,0 +1,254 @@
+swagger: "2.0"
+info:
+  version: 1.0.0
+  title: Proxy PagoPA
+  description: Proxy for PagoPA payment services
+basePath: /api/v1
+schemes:
+  - https
+paths:
+  "/payment-requests/{rptId}":
+    x-swagger-router-controller: PaymentController
+    get:
+      operationId: getPaymentInfo
+      summary: Get Payment Info
+      description: Retrieve information about a payment
+      parameters:
+        - name: rptId
+          in: path
+          required: true
+          schema:
+            "$ref": "#/definitions/RptIdFromString"            
+      responses:
+        '200':
+          description: Payment information retrieved
+          schema:
+            "$ref": "#/definitions/PaymentRequestsGetResponse"
+          examples:
+            application/json:
+              importoSingoloVersamento: 200,
+              codiceContestoPagamento: "ABC123"
+        "400":
+          description: Invalid input or PagoPA response
+        "500":
+          description: PagoPA services are not available or request is rejected
+  "/payment-activations":
+    x-swagger-drouter-controller: PaymentController
+    post:
+      operationId: activatePayment
+      summary: Activate Payment
+      description: Require a lock (activation) for a payment
+      parameters:
+        - in: body
+          name: body
+          schema:
+            $ref: "#/definitions/PaymentActivationsPostRequest"
+          x-examples:
+            application/json:
+              rptId: "DVCMCD99D30E611V044012345678901200"
+              importoSingoloVersamento: 200
+              codiceContestoPagamento: "ABC123"
+      responses:
+        '200':
+          description: Payment activation process started
+          schema:
+            "$ref": "#/definitions/PaymentActivationsPostResponse"
+          examples:
+            application/json:
+              importoSingoloVersamento: 200
+        "400":
+          description: Invalid input or PagoPA response
+        "500":
+          description: PagoPA services are not available or request is rejected
+  "/payment-activations/{codiceContestoPagamento}":
+    x-swagger-router-controller: PaymentController
+    get:
+      operationId: getActivationStatus
+      summary: Get Activation status
+      description: Check the activation status to retrieve the paymentId
+      parameters:
+        - name: codiceContestoPagamento
+          in: path
+          required: true
+          description: The transaction id for PagoPA
+          schema:
+            $ref: "#/definitions/CodiceContestoPagamento"
+      responses:
+        '200':
+          description: Payment information
+          schema:
+            $ref: "#/definitions/PaymentActivationsGetResponse"
+          examples:
+            application/json:
+              idPagamento: "123455"
```

Comment:
is this a possible pattern for idPagamento ?

Reply:
Maybe. It's defined into specs ad: WithinRangeString(1, 35)